README.md

```diff
@@ -87,59 +53,177 @@ dojo "gradle test jar"
+
+
+## Why was Dojo created? Dojo benefits
+
+Docker and containerization has revolutionized application lifecycle by creating a common standard. Dojo uses Docker to provide a similar standard for environments, providing environments as code. Dojo allows environments to be versioned, released as a Docker image and used in the Infrastructure As Code manner.
+
+Why was Dojo created? Dojo benefits:
+* to **shorten setup time of development environment** to near-zero
+* to **have environment parity** - environment consistent between multiple developers or between a developer and a CI server (fixes **works on my machine** problem)
+* to **have easily reproducible environments** (if your environment changed and you experience configuration drift, you can just recreate a Dojo Docker container)
+* to **treat environment as code** and lock it in source control of a project. The environment of a project is thus specified in one place, in [Dojofile](#dojofile)
+* to **execute commands in docker** much more easily than with bare `docker run`. Dojo takes care of many little details. (See [lower why](#Why-not-just-docker-run))
+
+Problems solved by Dojo and surrounding practices:
+1. **Works on my machine** - with Dojo, each host (a developer's workstation or a CI agent) gets a **consistent, reproducible** environment delivered by a docker image.
+1. **Configuration management** - with Dojo, any development environment is built from a [Dockerfile](https://docs.docker.com/engine/reference/builder/), which is versioned and released as a docker image.  Therefore the provisioning of the environment is stated in several commands, providing both documentation and a setup script in the single source of truth.
+1. Tools on CI agents or developer's laptop over time may mismatch with project's required setup - With Dojo the environment is pulled as docker image just before the operation to be executed, therefore **hosts don't need to be provisioned repeatedly to keep up with changing projects**.
+
+## Who is it for?
+
+1. **as-code practitioners** - If you like *as-code philosophy* (infrastructure-as-code, pipelines-as-code) then you'll like Dojo, which provides **development and operations environment as code**.
+1. Consultants, DevOps Engineers, **anyone jumping between projects and languages often**. With Dojo, getting the *right environment* for a project takes as long as the `docker pull`.
+1. **Polyglot software houses**, especially where CI agents are shared by a variety of projects.
+1. If you are already running builds with docker, then Dojo will make your scripts shorter and your life easier.
+
+By adopting Dojo you would be shifting responsibility for the development environment closer to the developers.
+A dojo docker image becomes a contract of what is a **correct environment** for a project **at a particular commit** (see [Dojofile](#dojofile)).
+
+# Table of contents
```

Comment:
It would be convenient to have this higher on the page.

Also, it does not mention the 3 examples at all.